stdlib/seq.mc

```diff
@@ -14,14 +14,26 @@ let foldl = fix (lam foldl. lam f. lam acc. lam seq.
 )
 let foldl1 = lam f. lam l. foldl f (head l) (tail l)
 
+let foldr = fix (lam foldr. lam f. lam acc. lam seq.
+    if null seq
+    then acc
+    else f (foldr f acc (tail seq)) (head seq)
+)
+
+let rev = lam seq.
+    let f = lam acc. lam x. cons x acc in
+    foldl f [] seq
+
+let foldr1 = lam f. lam seq. foldl1 f (reverse seq)
```

Comment:
Not sure if `foldr f (head seq) (tail seq)` would be faster (no reverse, but also not tail-recursive).

Reply:
If you do (head [1,2,3]) you get 1 but when folding from right you want 3 to be the first value of the accumulator right?

Reply:
You're right! Maybe you have to implement it from scratch to get rid of the reverse. Let's just leave as is then!